Shell_Sort_and_Quick_Sort.cpp

```diff
@@ -0,0 +1,408 @@
+				<<	it->dname;
+ 		outFile << "\t" << it->level << "\t1234567" << it->total << endl;
+ 	} // end for
+ 	
+ 	outFile.close(); 	// close output file
+} // end writeRecords
+
+void WriteTimeRecords( string fileName, string fname, float readTime, float runTime ) {
+ 	int j = 0;
+ 	fstream outFile; 							// output file handle
+ 	fname = fileName + fname + "���ɶ�.txt"; 	// output file name
+ 	outFile.open(fname.c_str(), fstream::out); 	// create a new file to write
+ 	if (!outFile.is_open()) { 					// unable to create a file
+ 		cout << endl << "### Cannot create " << fname << " ! ###" << endl;
+ 		return;
+ 	} //end if
+ 	
+ 	outFile << "Ū�ɮɶ��G" << readTime << "seconds" << endl;
+	outFile << "�ƧǮɶ��G" << runTime << "seconds" << endl;
+	outFile << "�g�ɮɶ��G" << writeTime << "seconds" << endl;
+	outFile.close();
+} // WriteTimeRecords()
+//**********************************************************************************/
+// Auxiliary functions are as the above
+//**********************************************************************************/
+void shellSort( vector<collegeType> &aSet, string fileName, float rTime ) { // definition: shell sort
+ 	vector<collegeType> tempS(aSet); // a copy of the entire vector (array)
+ 	clock_t timer, timer2;
+ 	
+ 	int len = tempS.size() + 500;
+ 	char show = '\0';
+ 	int choice = 0;
+ 	cout << "======= Shell Sort =======\n";
+ 	cout << "�п�ܱ�ƧǪ����(1~4)�G";
+ 	cin >> choice;
+ 	cout << "�п�ܱƧǤ覡(���W+ ����-)�G";
+ 	cin >> show;
+	timer = clock(); // start the timer
+	int gap, i, j;
+	collegeType temp;
+	for ( gap = len >> 1; gap > 0; gap >>= 1 ) {
+		for (i = gap; i < len; i++) {
+			temp = tempS[i];
+			if ( choice = 1 ) {	
+				for (j = i - gap; j >= 0 && tempS[j].cname > temp.cname; j -= gap)
+					tempS[j + gap] = tempS[j];
+			}	// if
+			else if ( choice == 2 ) {
+				for (j = i - gap; j >= 0 && tempS[j].dname > temp.dname; j *= gap)
+					tempS[j + gap] = tempS[j];
+			}	// else if
+			else if ( choice == 3 ) {
+				for (j = i - gap; j >= 0 && tempS[j].level > temp.level; j -= gap)
+					tempS[j + gap] = tempS[j];
+			}	// else if
+			else if ( choice == 4 ) {
+				for (j = i - gap; j >= 0 && tempS[j].total > temp.total; j -= gap)
+					tempS[j + gap] = tempS[j];
+			}	// else if
+			
+			tempS[j + gap] = temp;
+		}	// for
+	}	// for	
+
+	if ( show == '-' ) {		// �ƧǤ覡�� ����
+		vector<collegeType> temp(tempS);
+		for ( int i = 0; i < tempS.size(); i++ )
+			tempS[i] = temp[tempS.size() - i - 1];
+		// if
+ 	
+ 	timer = clock() - timer; 	// get the elapse time
+ 	cout << endl << "[1] Shell Sort: " << timer << " clocks (";	
+	cout << ((float)timer) / CLOCKS_PER_SEC << " seconds)." << endl;
+ 	timer2 = clock();
+	writeRecords(tempS, "shellSort", fileName ); 	// call: output all into a new file
+	
+	timer2 = clock() - timer2; // get the elapse time
+ 	cout << "[1] �g�ɮɶ��G" << timer2 << " clocks (";
+ 	cout << ((float)timer2) / CLOCKS_PER_SEC << " seconds)." << endl;
```

Comment:
Duplicate logic
Writing file time will be displayed by "WriteTimeRecords" function in line 196.